modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinServerCodegen.java

```diff
@@ -106,6 +110,9 @@ public class KotlinServerCodegen extends AbstractKotlinCodegen implements BeanVa
     public KotlinServerCodegen() {
         super();
 
+        // Enable proper oneOf/anyOf discriminator handling for polymorphism
+        legacyDiscriminatorBehavior = false;
```

Comment:
Since `oneOf` and Polymorphism weren't supported at all in the `kotlin-server` generator before, I consider this to be a non-breaking change, even though the default is set to `true` in `DefaultCodegen`. Let me know if you think otherwise.